fsck/fsck.c

```diff
@@ -668,7 +668,7 @@ static int exfat_de_iter_get(struct exfat_de_iter *iter,
 			return EOF;
 		} else if (ret <= 0) {
 			exfat_err("failed to read a cluster. %d\n", ret);
-			return ret;
+			return -ret;
```

Comment:
can not understand.. ret is already negative or zero.

Reply:
This function returns true/1 on failure
By returning ret here since it's negative the if checks fail, --=+ so we solve the issue.
Only thing is 0 which will still fail, might as well just return 1 or something?

Reply:
> This function returns true/1 on failure

This function returns negative integer if it fails, otherwise returns 0.

> By returning ret here since it's negative the if checks fail, --=+ so we solve the issue.
> Only thing is 0 which will still fail, might as well just return 1 or something?

Sorry, I can't understand it. Could you explain it again?

Reply:
Sorry you're right.
But with ret = 0 we return success while it actually failed.
Should the check be < 0 instead?